File: Runtime/InitialConditionsSpatialMap.vb
'************************************************************************************
' ST-Sim: A .NET class library for developing state-and-transition simulation models.
'
' Copyright © 2007-2017 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.
'
'************************************************************************************

Imports SyncroSim.Common

Class InitialConditionsSpatialMap

    Private m_HasItems As Boolean
    Private m_Map As New SortedKeyMap1(Of InitialConditionsSpatial)(SearchMode.ExactPrev)

    Public Sub New(ByVal ics As InitialConditionsSpatialCollection)

        For Each t As InitialConditionsSpatial In ics
            Me.AddICS(t)
        Next

    End Sub

    Private Sub AddICS(ByVal ics As InitialConditionsSpatial)

        Me.m_Map.AddItem(ics.Iteration, ics)

        Me.m_HasItems = True

    End Sub

    Public Function GetICS(ByVal iteration As Nullable(Of Integer)) As InitialConditionsSpatial

        If (Not Me.m_HasItems) Then
            Return Nothing
        End If

        Dim l As InitialConditionsSpatial = Me.m_Map.GetItem(iteration)

        Return l

    End Function

End Class

File: Runtime/InitialConditionsSpatialMap.cs
// ST-Sim: A SyncroSim Module for the ST-Sim State-and-Transition Model.
// Copyright © 2007-2018 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.

using SyncroSim.Common;

namespace SyncroSim.STSim
{
    internal class InitialConditionsSpatialMap
    {
        private bool m_HasItems;
        private SortedKeyMap1<InitialConditionsSpatial> m_Map = new SortedKeyMap1<InitialConditionsSpatial>(SearchMode.ExactPrev);

        public InitialConditionsSpatialMap(InitialConditionsSpatialCollection ics)
        {
            foreach (InitialConditionsSpatial t in ics)
            {
                this.AddICS(t);
            }
        }

        private void AddICS(InitialConditionsSpatial ics)
        {
            this.m_Map.AddItem(ics.Iteration, ics);

            this.m_HasItems = true;
        }

        public InitialConditionsSpatial GetICS(int? iteration)
        {
            if (!this.m_HasItems)
            {
                return null;
            }

            InitialConditionsSpatial l = this.m_Map.GetItem(iteration);

            return l;
        }
    }
}
